### PR TITLE
Issue #3305677: Remove destination in group invites url

### PR DESCRIPTION
**Pull request by NickDaelemans**
Because of the app's architecture the group invitation urls are not working. By removing the destination logic and using the url directly and let Drupal handle the login logic, this will work for both app and desktop.

## Problem
When sending group invitation mails, we include the accept url for the invite. We are sending the user to /user/login page with the accept url as a destination. However, this interferes with the app.

When you are invited to a private group, you get a mail, which contains a link to the new group. If you click on the link, you accept the invite. The link is going to /user/login with the accept url as a destination (https://example.com/user/login?destination=/social-group-invite/{group_id}/accept).

This doesn't work on the app, because all user/login url's are forced to open in the browser. From the moment you log in, you are redirected to the app, but it doesn't show you the accept url anymore, because it has lost that context. If you we send the user directly to the app (by not using /user/login, but the accept url directly, it will open the app (and if necessary) redirect the user to the browser to login. If the user is already logged in, it will show in the app.

## Solution
Change the url, so it doesn't contain the destination part anymore. Use the accept url as url for the email. Drupal will handle the login logic.

## Issue tracker
* https://www.drupal.org/project/social/issues/3305677
* https://getopensocial.atlassian.net/browse/APP-364

## How to test
- [x] Create a private group
- [x] Create a test user
- [x] Receive an invitation to any private group to the new user
- [x] If you don't receive a mail, try running cron
- [x] Open the email with an invitation
- [x] Tap the button

### Before merge
- [x] Code/peer review is completed
- [x] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [x] All automated tests are green
- [x] Functional/manual tests of the acceptance criteria are approved
- [x] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [x] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [x] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [x] This pull request has all required labels (team/type/priority)
- [x] This pull request has a milestone
- [x] This pull request has an assignee (if applicable)
- [x] Any front end changes are tested on all major browsers
- [x] New UI elements, or changes on UI elements are approved by the design team
- [x] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
*[Required if new feature, and if applicable] If this Pull Request makes visual changes then please include some screenshots that show what has changed here. A before and after screenshot helps the reviewer determine what changes were made.*

## Release notes
*[Required if new feature, and if applicable] A short summary of the changes that were made that can be included in release notes.*

## Change Record
*[Required if applicable] If this Pull Request changes the way that developers should do things or introduces a new API for developers then a change record to document this is needed. Please provide a draft for a change record or a link to an unpublished change record below. Existing change records can be consulted as example. Please provide a draft for a change record or a link to an unpublished change record below. [Existing change records](https://www.drupal.org/list-changes/social) can be consulted as example.*

## Translations
*[Optional]Translatable strings are always extracted from the latest development branch. To ensure translations remain available for platforms running older versions of Open Social the original string should be added to `translations.php` when it's changed or removed.*
- [ ] Changed or removed source strings are added to the `translations.php` file.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3077-n8wgrmqzhdbxljkmot7kgqi2ylujrytc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6305ed1df08423f9ad016932

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3077-n8wgrmqzhdbxljkmot7kgqi2ylujrytc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6305ed1df08423f9ad016932

File: modules/social_features/social_group/modules/social_group_invite/social_group_invite.module [hunk 1]
```diff
@@ -706,16 +705,14 @@ function social_group_invite_tokens_alter(array &$replacements, array $context,
       $context["tokens"]["register_link"] === '[group_content:register_link]'
     ) {
       // For new users for register_link token.
-      $link = $replacements["[group_content:register_link]"];
-      $replacements["[group_content:register_link]"] = "{$link}&{$destination}";
+      $replacements["[group_content:register_link]"] = $group_invite_url;
```
Comment (zanvidmar) [inline modules/social_features/social_group/modules/social_group_invite/social_group_invite.module:708]: This change is causing regression.

As admin if you send email invitation to user without account it is expected that invitation link includes user registration.

For example: 

If flexible group is visible to community or public the link should be something like:
https://example.com/user/register?invitee_mail={invite_token}&destination=/social-group-invite/{invite_id}/accept but with given fix it is https://example.com/social-group-invite/{invite_id}/accept and this is wrong

If flexible group is visible to members only the link should be something like:
https://example.com/user/register?invitee_mail={invite_token}&destination=/my-invites but with given fix it is https://example.com/my-invites and this is wrong

**Comment by zanvidmar** [comment]
@NickDaelemans please remove change in [phpstan-baseline.neon](https://github.com/goalgorilla/open_social/pull/3077/files#diff-995edee38ad4f8387e58ebd52c31bcc04c56cc2448d331b1cf5e0b35c57b9efa) and rebase to main as this change has been already pushed. Also please check if all automatic checks are successful. Thank you :)

File: modules/social_features/social_group/modules/social_group_invite/social_group_invite.module (hunk 1 above)
Comment (NickDaelemans) [reply modules/social_features/social_group/modules/social_group_invite/social_group_invite.module:718]: That's a great find, thanks! I updated the code, so now it uses the destination in the register link only.

Comment (zanvidmar) [reply modules/social_features/social_group/modules/social_group_invite/social_group_invite.module]: Tnx for update Nick, links have now the expected pattern.

**Comment by navneet0693** [comment]
@NickDaelemans I am not sure why the labels changes are not being picked up correctly by mergeables. I will take a look later for fixing this. Please feel free to merge this PR once ready.

**Comment by zanvidmar** [comment]
Except for "Mergeable: Valid labels" which is falsely failing, everything looks good 👍  Thank you Nick!

**Review by zanvidmar — APPROVED** [review]

**Comment by robertragas** [comment]
Cherry picked to 11.3.x

**Pull request merged**